### ISSUE TITLE
Date Support

### ISSUE THREAD

**Issue by magaton** [neo4j/graphql#244]
I have a node Month with properties of type Date in Neo4j
And my Graphql type is:

```
type Month {
   id: String!
   name: String!
   type: String!
   startDate: Date!
   endDate: Date!
   status: String
}

```
query:

```
query {
    months (where: { name_IN: ["2020/03", "2020/04"] }) {
        id
        name
        startDate
        endDate 
        status
    }
}
```

I want to get back  `2020-03-01` as startDate

Instead, I am getting:
```
"startDate": {
          "year": {
            "low": 2020,
            "high": 0
          },
          "month": {
            "low": 3,
            "high": 0
          },
          "day": {
            "low": 1,
            "high": 0
          }
        }

```

In neo4j-graphql-js I was able to specify `formatted` property on Date, now it is not possible.
This is a real blocker for me

**Comment by darrellwarde** [comment]
Thanks for this @magaton!

Also adding an important detail you sent from Discord:

> Now when I change schema:
> Date -> DateTime
> I am getting:
> ```
> "message": "Failed to invoke function `apoc.date.convertFormat`: Caused by: java.time.format.DateTimeParseException: Text '2020-03-01' could not be parsed at index 10", 
> ```

Because `date` and `datetime` are treated a bit differently in Neo4j, we likely need to also introduce a `Date` type here - I also think it will be good to give this option to developers, and it removes any concerns regarding timezones.

**Comment by jaedag** [comment]
Also encountering this bug. Hoping it gets fixed soon!

**Comment by jaedag** [comment]
I see that this issue has been closed, yet the bug has not been fixed even in the new release package. Is there any word on what the plan is for resolving this?

**Comment by darrellwarde** [comment]
> I see that this issue has been closed, yet the bug has not been fixed even in the new release package. Is there any word on what the plan is for resolving this?

Hey @jaedag, sorry for the slow reply on this one, closed issues and all that!

There is a new `Date` scalar which fixes this - it is not a fix for the `DateTime` scalar. This was released in 1.0.3, but it hasn't made it into a 2.0.0 release yet.
